src/kernel/task.zig

```diff
@@ -163,21 +162,24 @@ pub const Task = struct {
 /// Return: u32
 ///     A new PID.
 ///
-fn allocatePid() PidBitmap.IndexType {
+fn allocatePid() usize {
     return all_pids.setFirstFree() orelse panic(@errorReturnTrace(), "Out of PIDs\n", .{});
 }
 
 ///
 /// Free an allocated PID. One must be allocated to be freed. If one wasn't allocated will panic.
 ///
 /// Arguments:
-///     IN pid: u32 - The PID to free.
+///     IN pid: usize - The PID to free.
 ///
-fn freePid(pid: PidBitmap.IndexType) void {
-    if (!all_pids.isSet(pid)) {
+/// Error: BitmapError.
+///     OutOfBounds: The index given is out of bounds.
+///
+fn freePid(pid: usize) bitmap.BitmapError!void {
+    if (!(try all_pids.isSet(pid))) {
         panic(@errorReturnTrace(), "PID {} not allocated\n", .{pid});
     }
-    all_pids.clearEntry(pid);
+    all_pids.clearEntry(pid) catch unreachable;
```

Comment:
Given this function now can return an error, can `try all_pids.clearEntry(pid)` instead of `catch unreachable`.
But perhaps this function shouldn't return an error and just panic as would be used in `defer` statements, removing the need for catching the errors in defer statements.

Reply:
I think I agree with making it `try` instead :+1: